twoopstracker/twoops/views.py

```diff
@@ -389,7 +390,7 @@ def perform_create(self, serializer):
 
 class AccountsList(generics.RetrieveUpdateDestroyAPIView):
     queryset = TwitterAccountsList.objects.all()
-    serializer_class = TwitterAccountsListSerializer
+    serializer_class = TwitterAccountsListsSerializer
```

Comment:
Why not just remove the get accounts stuff from [`TwitterAccountsListSerializer`](https://github.com/CodeForAfrica/TwoopsTracker/blob/a509a053b36a721479edbbbaa59837d46c0bd74d/twoopstracker/twoops/serializers.py#L75)?

Reply:
Because we still need the data being represented from that serializer in other places e.g While processing files for download e.g [here](https://github.com/CodeForAfrica/TwoopsTracker/blob/ft-add-list-detail-view/twoopstracker/twoops/views.py#L136)

Reply:
Still, the right approach would be:

1. Continue to use TwitterAccountsListSerializer here, and
2. Add a check [before](https://github.com/CodeForAfrica/TwoopsTracker/blob/2b4a2a461ffc281e55afdc568b53a1f8a10860db/twoopstracker/twoops/serializers.py#L97) calling `get_accounts` to see if the request was for downloading (I believe `self.context['request'].GET.get("download")` is available in the Serializer class as well).

This way there are no surprises, `TwitterAccountsList` is serialized by `TwitterAccountsListSerializer`, and `TwitterAccountsLists` is serialized by `TwitterAccountsListsSerializer`

Reply:
This has been updated @kilemensi